### ISSUE TITLE
Critical v3.1.1 Play and Play and debug

### ISSUE THREAD

**Issue by coldestheart** [BabylonJS/Editor#129]
Yesterday was started using babylon editor 3.1.1 Desktop (Win10), when press Play scene didnt start, i thought ok, maybe my fault. I reopen editor with intro scene, press play and error was same.
Then i clicked Play and Debug ----

`node_modules/babylonjs/babylon.max.js:106113 BJS - [21:31:01]: BABYLON.CustomEditorMaterial not found, you may have missed an import.
./Misc/logger.ts.Logger._WarnEnabled @ node_modules/babylonjs/babylon.max.js:106113
node_modules/babylonjs/babylon.max.js:106123 BJS - [21:31:01]: Unable to load from file:scene389e904d-c588-4cbd-bd08-614564dc3d48.babylon: loadAssets of unknown
	Lights:
		Name: Spot Light, type: Spot
	Materials:
		Name: Amiga
		Name: default material
		Name: Wood
		Name: Sphere PBR
		Name: Ground
		Name: Documentation
		Name: Skybox
		Name: Custom material
./Misc/logger.ts.Logger._ErrorEnabled @ node_modules/babylonjs/babylon.max.js:106123
preview.html:8 Uncaught SyntaxError: Unexpected identifier
    at Function.Extension.AddScript (build/src/extensions/extension.js:22)
    at MaterialEditorExtension.createMaterial (build/src/extensions/material-editor/material-editor.js:65)
    at eval (build/src/extensions/material-editor/material-editor.js:104)
    at Array.forEach (<anonymous>)
    at MaterialEditorExtension.onApply (build/src/extensions/material-editor/material-editor.js:104)
    at Function.Extensions.ApplyExtensions (build/src/extensions/extensions.js:43)
    at preview.html:109
    at FileReader.reader.onload (node_modules/babylonjs/babylon.max.js:110469)
build/src/extensions/material-editor/material-editor.js:66 Uncaught TypeError: EDITOR.MaterialCreator.Constructors[id] is not a function
    at MaterialEditorExtension.createMaterial (build/src/extensions/material-editor/material-editor.js:66)
    at eval (build/src/extensions/material-editor/material-editor.js:104)
    at Array.forEach (<anonymous>)
    at MaterialEditorExtension.onApply (build/src/extensions/material-editor/material-editor.js:104)
    at Function.Extensions.ApplyExtensions (build/src/extensions/extensions.js:43)
    at preview.html:109
    at FileReader.reader.onload (node_modules/babylonjs/babylon.max.js:110469)`

Any way to fix this? Help please Like yours Editor but it too buggy(

**Comment by julien-moreau** [comment]
Hi @coldestheart! Thanks for using the Editor :)
You are experiencing the same issue as this one: https://forum.babylonjs.com/t/saving-and-exporting-into-a-web-project-from-the-babylon-editor/3574/11

Can you open the tool "Materials Editor" and remove the material that have been created? This fixes the issue.

The Material Editor tool will disappear as it is a fail and will be replaced by a node based material editor. So you should not re-open it :)

I'm releasing a fix ASAP which will be 3.1.2

**Comment by coldestheart** [comment]
It works, thanks, I really appreciate it, look forward to 3.1.2!

**Comment by julien-moreau** [comment]
Should be released really ASAP. I keep this issue open and will tell you once released :)
You can download the preview version here which removed the material editor tool: http://editor.babylonjs.com/electron/preview/BabylonJS Editor Setup 3.1.1.exe

Don't forget to remove the material and re-save your project before opening with the new preview version and it will work :)

**Comment by julien-moreau** [comment]
Sorry, copied the wrong value: http://editor.babylonjs.com/electron/preview/BabylonJS%20Editor%20Setup%203.1.1.exe

**Comment by coldestheart** [comment]
Thanks downloading

**Comment by coldestheart** [comment]
in the future please add reset Babylon Editor button ( for interface and all saved data ), sometimes inteface going crazy, and reinstalling didnt help. thank you in advance

**Comment by julien-moreau** [comment]
Adding as a new issue :)

**Comment by julien-moreau** [comment]
@coldestheart  you can find the latest build here: http://editor.babylonjs.com/electron/preview/BabylonJS%20Editor%20Setup%203.1.2-alpha.0.exe

This includes your latest feedbacks :)

**Comment by coldestheart** [comment]
Its аmazing, thanks for your work!

**Comment by coldestheart** [comment]
this.mesh.rotation.[ ] works only after you manually rotate object in inspector.

**Comment by coldestheart** [comment]
Enable physics break rotation animation, and position in inspector stop working

**Comment by coldestheart** [comment]
scene -> clear color dont accepting in play or play and debug

**Comment by julien-moreau** [comment]
For the clear color will check that and fix. I plan to release the 3.1.2 this weekend or tomorrow.

For the physics part I don’t reproduce here so I’m digging to understand the problem.

For the rotation part, do you prefer working with vectors or quaternions? I can add a way to configure the scene to export only quaternions or only vectors. The same option you can find in the BabylonJS exporter of 3ds Max. Do you agree?

**Comment by coldestheart** [comment]
I prefer vectors, and yes i think this option coud be very usefull, but does it means that mesh.rotation will be changed?

**Comment by julien-moreau** [comment]
Added to my todolist then :)
Nope, that means the rotation quaternions (if exist) will be translated to vectors 3D at the export or vectors to quaternions according to the selected setting

Do you agree with that?

**Comment by coldestheart** [comment]
Its pretty good, thank you

**Comment by julien-moreau** [comment]
@coldestheart added the option to export only euler angles or quaternions
* I can't reproduce the physics problem :(. How to have this bug? Just enabling physics and applying a physics impostor? Which kind of impostor?
* Looks like fixing the rendering pipleline will require me to fix the pipeline in babylonjs
* For the clear color I can't reproduce. Have you added a behavior and forget to remove it (Behavior Code Editor)?

Will share you the latest alpha version this evening before I officially release

**Comment by julien-moreau** [comment]
@coldestheart the alpha.1 is available here: http://editor.babylonjs.com/electron/preview/BabylonJS%20Editor%20Setup%203.1.2-alpha.1.exe

A menu has been added in the toolbar:
Project -> Project Settings...
You can now export euler angles instead of quaternions or export only quaternions instead of euler angles

**Comment by coldestheart** [comment]
Thanks, downloading. I'll try to give you more info about bugs tommorow, now i try your new version of bjs editor

**Comment by coldestheart** [comment]
About physics. take a look at this little part of [scene](https://drive.google.com/open?id=1wx9-Mpignp960-rkNULstfZBS4RSZmbD)

**Comment by coldestheart** [comment]
and may be u can add option to add animation value from the variable contains inside the script attached to object? it could be amazing

**Comment by coldestheart** [comment]
in graph editor cant add to board Events on pointer down, and other pointers but timeout works(

**Comment by julien-moreau** [comment]
@coldestheart, for the physics part, I got a warning while modifying impostors:
BJS - [22:17:20]: You must affect impostors to children before affecting impostor to parent.

This looks like the editor must be able to check that because this is needed by the physics engine in BabylonJS.

What do you prefer?
- The editor should apply the same kind of imposter you are applying to the child to all its parents?
- Hide the "physics" folder?
- Add a message to tell "you must set physics impostors to all parents before you can apply a "?

Adding multi-selection to my todolist. Before it is available, I suggest that you create a custom script that gets all the wanted meshes in order to apply an impostor using the code.

What do you mean by "and may be u can add option to add animation value from the variable that contained inside the script attached to object? it could be amazing"? Are you talking about the graph editor? If yes, sure I can add and very good idea!

For the graph editor part, I suggest that you open another issue and write the list of all the nodes you would like to use. This feature is still a POC and I would be very happy to populate the list of available nodes :)

**Comment by coldestheart** [comment]
I prefer --> Add a message to tell "you must set physics impostors to all parents before you can apply a "?;
About animation value, not in graph editor but in Animation editor, if you think that it is more necessary in the graph editor, I will agree with you.

**Comment by julien-moreau** [comment]
> and may be u can add option to add animation value from the variable that contained inside the script attached to object? it could be amazing

For this one, can you just elaborate a little bit more?
I'm focusing on that feature requests now :)

**Comment by coldestheart** [comment]
okey, for example in the script calculating offset for the camera apha and containing in alpha_offset variable, can u add option to create animation from current to alpha_offset

**Comment by coldestheart** [comment]
when i export scene i start getting strange error, after i try import scene.babylon into bjs editor play and play and debug stop working(and error shows), what im doing wrong? and what happens with the starsfield glowing on move (why its blinking unnatural); help plz it looks like a bug [this exported scene](https://drive.google.com/open?id=1ha0tsrLcRU2XRk_m50UUGAnNwxsO2jan)

**Comment by julien-moreau** [comment]
@coldestheart thanks for reporting! Just found and fixed! You can find the fix here: http://editor.babylonjs.com/electron/preview/BabylonJS%20Editor%20Setup%203.1.2-alpha.3.exe

For the animations feature, can you create another issue with some explications of what is exactly intended? Will be a pleasure to add the feature :)

**Comment by coldestheart** [comment]
Ok, i will create, but now when im enabling glow layer still present error in console
<a href="https://ibb.co/5nVxCPb"><img src="https://i.ibb.co/X3Tk6qc/image.png" alt="image" border="0"></a> does it normal?

**Comment by coldestheart** [comment]
#141

**Comment by julien-moreau** [comment]
Excellent thanks!
I'm trying to reproduce the problem with glow layer in babylonjs playground because it looks more like a parsing issue than an editor problem. Will come back with a fix ASAP :)
Thanks for the graph editor features, can't wait to integrate the new features!

**Comment by julien-moreau** [comment]
@coldestheart found and reported here: https://github.com/BabylonJS/Babylon.js/issues/6529
Anyway it looks like there is just an error in the console but the scene looks like it is working

**Comment by coldestheart** [comment]
thank for feedback man! I really appreciate it.

**Comment by julien-moreau** [comment]
It has been fixed by the bjs team! Will release once the fix has been released by the team :)

**Comment by julien-moreau** [comment]
Released :)
